File: src/main/java/com/chaottic/nbt/number/FloatTag.java
package com.chaottic.nbt.number;

import java.io.DataInput;
import java.io.IOException;

public record FloatTag(float f) implements NumberTag {
    @Override
    public byte getAsByte() {
        return (byte) f;
    }

    @Override
    public short getAsShort() {
        return (short) f;
    }

    @Override
    public int getAsInt() {
        return (int) f;
    }

    @Override
    public long getAsLong() {
        return (long) f;
    }

    @Override
    public float getAsFloat() {
        return f;
    }

    @Override
    public double getAsDouble() {
        return f;
    }

    public static FloatTag createFloatTag(DataInput dataInput) throws IOException {
        return new FloatTag(dataInput.readFloat());
    }
}

File: src/main/kotlin/com/chaottic/nbt/number/FloatTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.TagType
import java.io.DataInput

data class FloatTag(private val f: Float) : NumberTag {

	override fun asByte() = f.toInt().toByte()
	override fun asShort() = f.toInt().toShort()
	override fun asInt() = f.toInt()
	override fun asLong() = f.toLong()
	override fun asFloat() = f
	override fun asDouble() = f.toDouble()

	companion object : TagType<FloatTag> {

		override fun createTag(dataInput: DataInput) = FloatTag(dataInput.readFloat())
	}
}
